images/win/scripts/Installers/Install-AzureModules.ps1

```diff
@@ -36,6 +36,46 @@ foreach ($module in $modules)
         }
     }
 
+    if($null -ne $module.url)
+    {
+        $assets = Invoke-RestMethod $module.url
+    }
+
+    foreach ($version in $module.zipversions)
+    {
+        # Install modules from GH Release
+        if($null -ne $assets) 
+        {
+            $asset = $assets | Where-Object version -like $version `
+                         | Select-Object -ExpandProperty files `
+                         | Select-Object -First 1
+
+            Write-Host "Installing $($module.name) $version ..."
+            if ($null -ne $asset) {
+                Start-DownloadWithRetry -Url $asset.download_url -Name $asset.filename -DownloadPath $installPSModulePath
+            } else {
+                Write-Host "Asset was not found in versions manifest"
+                exit 1
+            }
+        }
+        # Install modules from powershell gallery
+        else 
+        {
+            $modulePath = Join-Path -Path $installPSModulePath -ChildPath "${moduleName}_${version}"
+            Write-Host " - $version [$modulePath]"
+            try
+            {
+                Save-Module -Path $modulePath -Name $moduleName -RequiredVersion $version -Force -ErrorAction Stop
+                Compress-Archive -Path $modulePath -DestinationPath "${modulePath}.zip"
+                Remove-Item $modulePath -Recurse -Force
+            }
+            catch
+            {
+                Write-Host "Error: $_"
+                exit 1
+            }
+        }
+    }
```

Comment:
As I understand, this code branch is needed for Azure and AzureRM modules that you don't distribute in `Azure/az-ps-module-versions`, am I right?

Considering the fact that new versions of Azure and AzureRM won't be released, can we compress them once and upload somewhere? If you don't want to use `Azure/az-ps-module-versions`, we can upload them to `https://vstsagenttools.blob.core.windows.net/tools`
We are trying to reduce image generation time so compressing the identical modules in every build looks excessive, especially because they won't be changed in future

Reply:
PR for uloading azure and azurerm modules to blob https://dev.azure.com/mseng/AzureDevOps/_git/CIPlat.Externals/pullrequest/601350

Reply:
Thank you!

Reply:
code changes to download from blob are done. please review